content/en/blog/_posts/2023-08-04-kubernetes-1.28-blog.md

```diff
@@ -0,0 +1,270 @@
+
+## Release Theme And Logo
+
+**Kubernetes v1.28: *Planternetes*
+
+The theme for Kubernetes v1.28 is *Planternetes*.
+
+{{< figure src="/images/blog/2023-08-15-kubernetes-1.28-blog/kubernetes-1.28.png" alt="Kubernetes 1.28 Planternetes logo" class="release-logo" >}}
+
+Each Kubernetes release is the culmination of the hard work of thousands of individuals from our community. The people behind this release are industry veterans, parents, others are students new to the ecosystem, contributing our unique experience to create a collective artifact with global impact.
+
+Much like a garden, our release has ever-changing growth, challenges and opportunities. This theme celebrates the meticulous care, intention and efforts to get the release to where we are today. Harmoniously together, we grow better.
+
+# What's New (Major Themes)
+
+## Add Support To Handle Non-graceful Node Shutdown
+    
+This feature allows stateful workloads to restart on a different node if the original node is shut down unexpectedly or ends up in a non-recoverable state perhaps due to hardware failure or unresponsive OS.
+
+This allows stateful workloads to failover to a different node successfully after the original node is shut down or in a non-recoverable state such as the hardware failure or broken OS.
+    
+The Graceful Node Shutdown introduces a way to detect a node shutdown and handle it gracefully. However, a node shutdown action may not be detected by Kubelet's Node Shutdown Manager, either because the command does not trigger the inhibitor locks mechanism used by Kubelet or because of a user error, i.e., the ShutdownGracePeriod and ShutdownGracePeriodCriticalPods are not configured properly.
+
+When a node is shutdown but not detected by Kubelet's Node Shutdown Manager, the pods that are part of a StatefulSet will be stuck in terminating status on the shutdown node and cannot move to a new running node. This is because Kubelet on the shutdown node is not available to delete the pods so the StatefulSet cannot create a new pod with the same name. If there are volumes used by the pods, the VolumeAttachments will not be deleted from the original shutdown node so the volumes used by these pods cannot be attached to a new running node. As a result, the application running on the StatefulSet cannot function properly. If the original shutdown node comes up, the pods will be deleted by Kubelet and new pods will be created on a different running node. If the original shutdown node does not come up, these pods will be stuck in terminating status on the shutdown node forever.
+
+This handles node shutdown cases that are not detected by Kubelet's Node Shutdown Manager. The pods will be forcefully deleted in this case, triggering the deletion of the VolumeAttachments, and new pods will be created on a new running node so that application can continue to function.
```

Comment:
If this is a graduation to stable, try:
```markdown
## Generally available: cleanup following non-graceful node shutdown

Kubernetes v1.20 introduced cleanup handling following a node shutdown that didn't happen cleanly.
That could be because of a node failure, a forced (hard) reboot, or because of a planned shutdown that
the kubelet's _node shutdown manager_ failed to detect. That can then have an impact on StatefulSets
with a Pod still marked as running on the affected node.

In Kubernetes v1.28, cleanup following a non-graceful node shutdown is generally available, graduating
from beta. Whenever an ungraceful node shutdown occurs, the lack of node heartbeats from the affected
kubelet signals the control plane to delete the affected Pods. That forcible deletion triggers further cleanup
including the deletion of volume attachment record for any storage (PersistentVolumeClaims) that the
Pod was using. The StatefulSet controller can then make a new, replacement Pod on a healthy node.

For more information, read [non-graceful node shutdown](/docs/concepts/architecture/nodes/#non-graceful-node-shutdown) in the Kubernetes documentation.
```
That text should get tech review from SIG Node / SIG Apps, ideally.

Reply:
the SIG-Storage leads the feature.

/cc @xing-yang @msau42

And this feature isn't automatically trriggered by the kubernetes itself, so cluster administators should enable it manually by setting the node with the taint `node.kubernetes.io/out-of-service=out-of-service=nodeshutdown:NoExecute` or `node.kubernetes.io/out-of-service=out-of-service=hardwarefailure:NoExecute`.

A little suggestion changes:

```
If a cluster administator taints the non-recoverable node with `node.kubernetes.io/out-of-service`, that forcible deletion triggers ...
```

Reply:
"cleanup following non-graceful node shutdown" -> We have been calling the feature "non-graceful node shutdown", versus the "graceful node shutdown" feature.
It was introduced as alpha in 1.24.
Yes, it is not automatic. User needs to apply the "out-of-service" taint on the shutdown node.

Reply:
> "cleanup following non-graceful node shutdown" -> We have been calling the feature "non-graceful node shutdown", versus the "graceful node shutdown" feature.

I know, but I've never liked it. Here's why:
- graceful node shutdown makes the kubelet shutdown more graceful
- “non-graceful node shutdown” **does not** make kubelet shut down less gracefully

The two features are unrelated.

Reply:
Hey folks, do we need to resolve any debates here? Please advise as I would like to resolve this today. Should I change it to what @sftim suggested for the description? thanks in advance

Reply:
Hi folks we are going to be consistent with the KEP and keep the non-graceful wording. Thanks all for your help we appreciate it.